spec/controllers/admin_controller_spec.rb

```diff
@@ -0,0 +1,30 @@
+require 'rails_helper'
+
+RSpec.describe AdminController, type: :controller do
+  describe '#authenticate_admin' do
```

Comment:
I've changed how this test was to implemented. The test now creates a mock controller that inherits from the admin controller. Then the tests test that controller.